src/Control/Effect/Pure.hs

```diff
@@ -1,50 +1,50 @@
 {-# LANGUAGE DeriveFunctor, EmptyCase, KindSignatures, MultiParamTypeClasses #-}
-module Control.Effect.Void
-( Void
+module Control.Effect.Pure
+( Pure
 , run
-, VoidC(..)
+, PureC(..)
 ) where
 
 import Control.Effect.Carrier
 import Data.Coerce
 
-data Void (m :: * -> *) k
+data Pure (m :: * -> *) k
   deriving (Functor)
 
-instance HFunctor Void where
+instance HFunctor Pure where
   hmap _ v = case v of {}
   {-# INLINE hmap #-}
 
-instance Effect Void where
+instance Effect Pure where
   handle _ _ v = case v of {}
   {-# INLINE handle #-}
 
 
 -- | Run an action exhausted of effects to produce its final result value.
-run :: VoidC a -> a
-run = runVoidC
+run :: PureC a -> a
+run = runPureC
 {-# INLINE run #-}
 
-newtype VoidC a = VoidC { runVoidC :: a }
+newtype PureC a = PureC { runPureC :: a }
```

Comment:
We can attach docs to the selector, and thus name the selector `run` instead of using a separate top-level declaration for it. On the other hand, this is pleasantly consistent, and I’m not sure if it’s really worth the four or so lines saved.